### PR TITLE
feat(filter-by): add strict mode; fixes #163

### LINKED ISSUE #163: filterBy doesn't have a strict mode
When using filterBy, I've noticed that it doesn't have a strict mode.

For example:

Say I have a select list that I'm using to filter my data by. And I have 2 items in there:

Hodgkin Lymphoma
Non-Hodgkin Lymphoma

If I select 'Hodgkin Lymphoma', I get hits containing 'Non-Hodgkin Lymphoma', because it contains 'Hodgkin Lymphoma' in the string match. There should be a mode or modifier that will require the match to be strict, and only match the strings when they are exact matches.

### PR DESCRIPTION
**Pull request by toddbranch**

### CODE REVIEW

File: src/_filter/collection/filter-by.js [hunk 1]
```diff
@@ -32,16 +32,19 @@ angular.module('a8m.filter-by', [])
...
           } else {
-            var propList = prop.replace(new RegExp('\\s', 'g'), '').split('+');
-            comparator = propList.reduce(function(prev, cur, index) {
-              return (index === 1) ? $parse(prev)(elm) + ' ' + $parse(cur)(elm) :
-                prev + ' ' + $parse(cur)(elm);
-            });
+            var propList = prop.replace(/\s+/g, '').split('+');
+            comparator = propList
+              .map(function(prop) { return $parse(prop)(elm); })
+              .join(' ');
```
Comment (toddbranch) [inline src/_filter/collection/filter-by.js]: Small refactor here.

File: test/spec/filter/collection/filter-by.js [hunk 2]
```diff
@@ -92,4 +92,41 @@ describe('filterByFilter', function() {
 
   });
 
+  it('should not coerce non-string/number properties', function() {
+    var users = [
+      { id: [1, 2], user: { first_name: 'foo', last_name: 'bar',  mobile: 4444 } }
+    ];
+
+    expect(filter(users, ['id'], 1)).toEqual([]);
+  });
```
Comment (toddbranch) [inline test/spec/filter/collection/filter-by.js]: Test for the `isString` / `isNumber` check

**Comment by a8m** [comment]
looks great @toddbranch, thx!

**Comment by a8m** [comment]
would you mind to update the readme file too?

**Comment by toddbranch** [comment]
Yeah no problem, I will submit PR later today.

**Pull request merged**